src/luarocks/argparse.lua

```diff
@@ -0,0 +1,2013 @@
+-- The MIT License (MIT)
```

Comment:
@hishamhm how would you like argparse brought in here? 
Vendoring it in at  src/luarocks/argparse.lua keeps working on luarocks itself as the same flow. But it's not obvious that this is a vendored dependency.

Other options:

  - vendor/argparse.lua (note that e.g. github knows not to count the top level vendor folder as part of a repository for stats purposes)
  - vendor/argparse/* as the whole argparse repository (brought in via `git subtree`)
  - vendor/argparse as a git submodule

Reply:
Tracked in #1067